### PR TITLE
Cache dependencies in travis ci

### PR DESCRIPTION
**Pull request by themattchan**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,2 +1,21 @@
...
-script: nix-build --attr proto3-suite-linux --arg enableDhall true release.nix
\ No newline at end of file
+
+script: nix-build --attr proto3-suite-linux --arg enableDhall true release.nix
+
+## Caching: https://github.com/travis-ci/travis-ci/issues/6604
+sudo: required
+
+install: |
+  if [ "$(ls -A $HOME/nix-cache)" ]; then
```
Comment (Gabriel439) [inline .travis.yml]: I know this code is taken from the above issue, but there are still a few things that I think we can fix:

```suggestion
  if [ -e ~/nix-cache ]; then
```

File: .travis.yml [hunk 2]
```diff
@@ -1,2 +1,21 @@
...
+    sudo rm -rf /nix/*
+    cp -a $HOME/nix-cache/* /nix
```
Comment (Gabriel439) [inline .travis.yml]:
```suggestion
    cp -a ~/nix-cache/* /nix
```

File: .travis.yml [hunk 3]
```diff
@@ -1,2 +1,21 @@
...
+  fi
+
+before_cache: |
+  if [ ! "$(ls -A $HOME/nix-cache)" ]; then
```
Comment (Gabriel439) [inline .travis.yml]:
```suggestion
  if [ ! -e ~/nix-cache ]; then
```

File: .travis.yml [hunk 4]
```diff
@@ -1,2 +1,21 @@
...
+    cp -a /nix/* $HOME/nix-cache
```
Comment (Gabriel439) [inline .travis.yml]:
```suggestion
    cp -a /nix/* ~/nix-cache
```

File: .travis.yml [hunk 5]
```diff
@@ -1,2 +1,21 @@
...
+  fi
+
+cache:
+  directories:
+    - $HOME/nix-cache
```
Comment (Gabriel439) [inline .travis.yml]:
```suggestion
    - ~/nix-cache
```

File: .travis.yml [hunk 6]
```diff
@@ -1,2 +1,21 @@
...
\ No newline at end of file
+
+script: nix-build --attr proto3-suite-linux --arg enableDhall true release.nix
+
+## Caching: https://github.com/travis-ci/travis-ci/issues/6604
+sudo: required
+
+install: |
+  if [ "$(ls -A $HOME/nix-cache)" ]; then
+    sudo rm -rf /nix/*
```
Comment (Gabriel439) [inline .travis.yml]:
```suggestion
    sudo rm -rf -- /nix/*
```

Comment (ixmatus) [reply .travis.yml]: TIL about getopts `--`.

**Pull request merged**